drivers: display: add HUB12 LED matrix driver

heronet opened the pull request:
Add driver for HUB12 interface monochrome LED matrix displays.

Features:
- 32x16 pixel resolution, 1-bit monochrome (PIXEL_FORMAT_MONO01)
- SPI-based data transfer with shift registers
- Configurable brightness control
- Thread-safe framebuffer access with semaphore

The driver implements the standard Zephyr display API

maass-hamburg commented on dts/bindings/display/hub12.yaml:
```diff
@@ -0,0 +1,34 @@
+  with additional GPIO pins for row address selection (PA, PB),
+  output enable (PE), and data latching (PLAT).
+
+compatible: "hub12"
```
Has to be vendor, device. Also use the specific chip, that is used, if there are multiple chips, use the one that this is based (the original not the clones) on.

maass-hamburg replied:
is there at least some documentation you could refer there then

heronet commented:
Hi @JarmouniA 
You reviewed a previous driver of mine. Care to take a look at this one when you get the time?

JarmouniA commented on dts/bindings/display/hub12.yaml:
```diff
@@ -0,0 +1,57 @@
+  - https://buildcircuits.com/blogs/scoreduino-modules/exploring-led-display-interfaces-from-hub75-to-spi-and-beyond
+  - https://www.iled.com/class/INNOVAEditor/assets/YeniDatasheets/2120-2124p10tekrenk.pdf
+
+compatible: "hub12"
```
```suggestion
compatible: "zephyr,hub12"
```

heronet committed:
dts: bindings: display: add HUB12 LED matrix binding

Add device tree binding for HUB12 interface monochrome LED matrix
displays (32x16 pixels).

The HUB12 interface uses SPI for data transfer to shift registers,
with additional GPIO pins for row address selection (PA, PB),
output enable (PE), and data latching (PLAT).

Signed-off-by: Siratul Islam <sirat4757@gmail.com>

heronet committed:
drivers: display: add HUB12 LED matrix driver

Add driver for HUB12 interface monochrome LED matrix displays.

Features:
- 32x16 pixel resolution, 1-bit monochrome (PIXEL_FORMAT_MONO01)
- SPI-based data transfer with shift registers
- Configurable brightness control
- Thread-safe framebuffer access with semaphore

The driver implements the standard Zephyr display API

Signed-off-by: Siratul Islam <sirat4757@gmail.com>

sylvioalves commented on drivers/display/display_hub12.c:
```diff
@@ -0,0 +1,366 @@
+#define HUB12_PIXELS_PER_BYTE 8
+
+/* Brightness control parameters */
+#define HUB12_PWM_FREQ           1000
```
Is it used anywhere?

heronet commented on drivers/display/display_hub12.c:
```diff
@@ -0,0 +1,366 @@
+	struct hub12_data *data = CONTAINER_OF(work, struct hub12_data, scan_work);
+	const struct hub12_config *config = data->dev->config;
+
+	hub12_scan_row(data, config);
```
Although in my testing there was no issue but a semaphore could be used here. I'll add it in a future pr as it is merged already
